pay/contracts/subscription/HectorSubscription.sol

```diff
@@ -0,0 +1,526 @@
+            if (
+                amount == 0 ||
+                amount > IERC20(token).balanceOf(address(this)) ||
+                amount > refundOf[to][token]
+            ) revert INVALID_AMOUNT();
+
+            unchecked {
+                refundOf[to][token] -= amount;
+            }
+
+            IERC20(token).safeTransfer(to, amount);
+
+            emit Refunded(to, token, amount);
+        }
+    }
+
+    /* ======== VIEW FUNCTIONS ======== */
+
+    function allPlans() external view returns (Plan[] memory) {
+        return plans;
+    }
+
+    function getSubscription(address from)
+        external
+        view
+        returns (
+            uint256 planId,
+            uint48 expiredAt,
+            bool isCancelled,
+            bool isActiveForNow,
+            uint256 chargeAmount
+        )
+    {
+        Subscription memory subscription = subscriptions[from];
+        planId = subscription.planId;
+        expiredAt = subscription.expiredAt;
+
+        // cancelled subscription
+        if (planId == 0) {
+            return (0, expiredAt, true, block.timestamp < expiredAt, 0);
+        }
+
+        // before expiration
+        if (block.timestamp < expiredAt) {
+            isActiveForNow = true;
+        } else {
+            // after expiration
+            Plan memory plan = plans[planId];
+            uint256 count = (block.timestamp - expiredAt + plan.period) /
+                plan.period;
+            uint256 amount = plan.amount * count;
+            uint256 balance = balanceOf[from][plan.token];
+
+            if (balance >= amount) {
+                isActiveForNow = true;
+                chargeAmount = 0;
+            } else {
+                isActiveForNow = false;
+                chargeAmount = amount - balance;
+            }
+        }
+    }
+
+    /* ======== USER FUNCTIONS ======== */
+
+    function deposit(address _token, uint256 _amount) public nonReentrant {
+        balanceOf[msg.sender][_token] += _amount;
+
+        IERC20(_token).safeTransferFrom(msg.sender, address(this), _amount);
+
+        emit PayerDeposit(msg.sender, _token, _amount);
+    }
+
+    function createSubscription(uint256 _planId) public onlyValidPlan(_planId) {
+        Subscription storage subscription = subscriptions[msg.sender];
+
+        // check if no or expired subscription
+        if (subscription.planId > 0) {
+            syncSubscription(msg.sender);
+
+            if (subscription.expiredAt > block.timestamp)
+                revert ACTIVE_SUBSCRIPTION();
+        }
+
+        Plan memory plan = plans[_planId];
+
+        // Pay first plan
+        if (balanceOf[msg.sender][plan.token] < plan.amount)
+            revert INSUFFICIENT_FUND();
+
+        unchecked {
+            balanceOf[msg.sender][plan.token] -= plan.amount;
+        }
+
+        IERC20(plan.token).safeTransfer(treasury, plan.amount);
+
+        // Set subscription
+        subscription.planId = _planId;
```

Comment:
@bejaxer Please move line 336 & 337 above line 333.
Reason: State variables written after the call(s) - https://github.com/crytic/slither/wiki/Detector-Documentation#reentrancy-vulnerabilities